market/mk12/mk12.go

```diff
@@ -424,24 +424,28 @@ func (m *MK12) processDeal(ctx context.Context, deal *ProviderDealState) (*Provi
 		}, nil
 	}
 
-	// de-serialize transport opaque token
-	tInfo := &HttpRequest{}
-	if err := json.Unmarshal(deal.Transfer.Params, tInfo); err != nil {
-		return &ProviderDealRejectionInfo{
-			Reason: fmt.Sprintf("failed to de-serialize transport params bytes '%s': %s", string(deal.Transfer.Params), err),
-		}, nil
-	}
+	var headers []byte
+	var tInfo *HttpRequest
```

Comment:
This must be
```suggestion
	tInfo := &HttpRequest{}
```
Otherwise the unmarshal below will error/panic

Reply:
Fixed